mountpoint-s3/src/fstab.rs

```diff
@@ -0,0 +1,252 @@
+    type Error = clap::Error;
+
+    fn try_from(fstab_cli_args: FsTabCliArgs) -> Result<Self, Self::Error> {
+        let cli_arg_list = fstab_cli_args.into_cli_arg_list()?;
+
+        let mut cli_args = CliArgs::try_parse_from(cli_arg_list)?;
+        cli_args.foreground = true;
+        Ok(cli_args)
+    }
+}
+
+impl FsTabCliArgs {
+    /// Parse the args we've been given into an iterator of options to pass to the main CliArgs parser
+    /// Filters out arguments that aren't meant for Mountpoint, and add `--` to any argument that's not prefixed with `-`.
+    fn into_cli_arg_list(self) -> Result<Vec<String>, clap::Error> {
+        Self::validate_options(&self.options)?;
+
+        let cli_arg_list = [
+            env::args().nth(0).unwrap_or("mount-s3".to_string()),
+            self.bucket_name,
+            self.mount_point,
+        ]
+        .into_iter()
+        .chain(
+            self.options
+                .into_iter()
+                .filter(|option| Self::option_allowed(option))
+                .map(Self::rename_option),
+        )
+        .collect();
+        Ok(cli_arg_list)
+    }
+
+    /// Returns if an option is allowed for systemd/the fstab parser
+    /// The hard-coded options listed aren't relevant to Mountpoint, but means we can't add them in the future.
+    /// Options prefixed with `x-` are 'comments' in fstab, and can be ignored by us.
+    fn option_allowed(option: &str) -> bool {
+        // Same as from https://github.com/libfuse/sshfs/blob/ed0825440c48895b7e20cc1440bbafd8d9c88eb8/sshfs.c#L533-L538 with addition of `rw` argument
+        !(["auto", "noauto", "user", "nouser", "users", "_netdev", "rw"].contains(&option) || option.starts_with("x-"))
+    }
+
+    fn rename_option(option: String) -> String {
+        match option.as_str() {
+            "ro" => "--read-only".to_string(),
+            _ => format!("--{}", option),
+        }
+    }
+
+    fn validate_options(options: &[String]) -> Result<(), clap::Error> {
+        if options.iter().any(|x| x.starts_with("-")) {
+            return Err(Command::error(
+                &mut Self::command(),
+                ErrorKind::InvalidValue,
+                "Cannot prefix arguments with `--` when using fstab style arguments.",
+            ));
+        }
+        if options.iter().any(|x| x == "ro") && options.iter().any(|x| x == "rw") {
+            return Err(Command::error(
+                &mut Self::command(),
+                ErrorKind::ArgumentConflict,
+                "Cannot use 'ro' flag combined with 'rw' flag.",
+            ));
+        }
+        if let Some(disallowed_option) = options
+            .iter()
+            .find(|x| ["foreground", "read-only"].contains(&x.as_str()))
+        {
+            return Err(Command::error(
+                &mut Self::command(),
+                ErrorKind::InvalidValue,
+                format!(
+                    "Cannot use '{}' flag when using fstab style arguments.",
```

Comment:
So we only support `ro` for `--read-only` and disallow directly using `ready-only` right? Would be useful to note that in the error message

Reply:
Will edit in the next PR